OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -236,7 +233,7 @@ public virtual Barrel CheckFire(Actor self, IFacing facing, Target target)
 				Source = muzzlePosition(),
 				CurrentSource = muzzlePosition,
 				SourceActor = self,
-				PassiveTarget = attack.Info.AttackTargetCenter ? target.CenterPosition : target.Positions.PositionClosestTo(muzzlePosition()),
+				PassiveTarget = attack != null && attack.Info.AttackTargetCenter ? target.CenterPosition : target.Positions.PositionClosestTo(muzzlePosition()),
```

Comment:
The issue prompting my question above is that this then forces the equivalent for `AttackTargetCenter: True` for pillboxes here.